### ISSUE TITLE
rewriting Nat.zero and Nat.succ after induction(') tactic

### ISSUE THREAD

**Issue by robertylewis** [leanprover-community/mathlib4#1558]
This is either a feature request or a reminder for myself and @hrmacbeth to implement a feature for our courses.

Right now, the `induction` and `induction'` tactics produce goals with bare constructors in them. (This isn't surprising and mimics Lean 3.) For instance:

```lean
example (a b : ℕ) : a + b = 0 := by 
  induction a

/-
unsolved goals
case zero
b : ℕ
⊢ Nat.zero + b = 0

case succ
b n✝ : ℕ
n_ih✝ : n✝ + b = 0
⊢ Nat.succ n✝ + b = 0
-/
```

We'd love for teaching purposes for these goals to look like `0 + b = 0` and `n + 1 + b = 0`. (In Lean 3, `nat.zero` gets pretty printed as 0, so this appears to half-happen.) An easy way around it is to write a wrapper for induction that does a `simp only at *`, which will be enough for our courses. But I wanted to open this issue in case people had thoughts about a more principled change, or in case this was something worth working into the mathlib4 tactics.

**Comment by fgdorais** [comment]
An easy fix for that is to update the default eliminator to use notations:
```lean
@[eliminator] protected def Nat.recAux.{u} {motive : Nat → Sort u} (zero : motive 0) (succ : (n : Nat) → (ih : motive n) → motive (n+1)) : (t : Nat) → motive t
| 0 => zero
| n+1 => succ n (Nat.recAux zero succ n)

example (a b : Nat) : a + b = 0 := by 
  induction a

/-
unsolved goals
case zero
b : Nat
⊢ 0 + b = 0

case succ
b n✝ : Nat
ih✝ : n✝ + b = 0
⊢ n✝ + 1 + b = 0
-/
```
Unfortunately, this makes `cases a` behave funny. That can probably be fixed. (When Leo generously added that feature at my request long ago, I was so grateful I didn't feel I should nitpick the details.)

**Comment by hrmacbeth** [comment]
After @fgdorais ' comment I realised an even cheaper fix: use the custom induction principle `Nat.recAux` he suggests which refers to `n + 1` rather than `n.succ`, but rather than changing the default eliminator to this, just always invoke it explicitly with `induction ... using Nat.recAux`.

It's a bit wordy, but one can write a wrapper tactic for this, and it works fine for low-level teaching where the students aren't going to encounter other kinds of Lean `induction` and be confused by the difference.

Given the existence of two reasonable workarounds for this, I'm going to close the issue.
